browser/search_engines/search_engine_provider_service_factory.cc

```diff
@@ -32,7 +34,7 @@ KeyedService* InitializeSearchEngineProviderServiceIfNeeded(Profile* profile) {
 
   // Regardless of qwant region, tor profile needs controller to store
   // previously set search engine provider.
-  if (profile->IsTorProfile() && IsGuestProfile(profile)) {
+  if (brave::IsTorProfile(profile) && brave::IsOffTheRecordProfile(profile)) {
```

Comment:
I don't think we want to differentiate with OTR here. All Tor profiles should use `TorWindowSearchEngineProviderService`